pkg/chunk/purger/purger_test.go

```diff
@@ -414,9 +417,17 @@ func TestPurger_Metrics(t *testing.T) {
 	require.InDelta(t, float64(2*86400), testutil.ToFloat64(purger.metrics.oldestPendingDeleteRequestAgeSeconds), 1)
 	require.Equal(t, float64(2), testutil.ToFloat64(purger.metrics.pendingDeleteRequestsCount))
 
-	// start loop to process requests
+	// stop the existing purger
+	require.NoError(t, services.StopAndAwaitTerminated(context.Background(), purger))
```

Comment:
At the beginning of the test there's a deferred call to `purger.StopAsync()`. Given we're now terminating it here, I believe that call should be removed.